frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/FetchAllRemotesAction.java

```diff
@@ -0,0 +1,72 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getProject;
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getSelectedVcsRepository;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import git4idea.fetch.GitFetchResult;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.common.GitFetchSupportImpl;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class FetchAllRemotesAction extends AnAction implements DumbAware {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    Project project = getProject(anActionEvent);
+    if (GitFetchSupportImpl.fetchSupport(project).isFetchRunning()) {
+      anActionEvent.getPresentation().setEnabled(false);
+      anActionEvent.getPresentation().setDescription("Update is already running...");
+    }
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    LOG.debug("Performing");
+
+    Project project = getProject(anActionEvent);
+    Option<GitRepository> selectedVcsRepository = getSelectedVcsRepository(anActionEvent);
+
+    new Task.Backgroundable(project, "Fetching...", /* canBeCancelled */ true) {
+
+      @Nullable
+      GitFetchResult result = null;
+
+      @Override
+      public void run(ProgressIndicator indicator) {
+        result = GitFetchSupportImpl.fetchSupport(project).fetchAllRemotes(selectedVcsRepository.toJavaList());
+      }
+
+      @Override
+      public void onFinished() {
+        if (result != null) {
+          result.showNotification();
```

Comment:
Hmm I think `result` should be treated as volatile here... so even if `result` was non-null in `if`'s condition, there's no guarantee (at least theoretically) that it's still non-null in this line :/ Assign to a local var maybe?

Reply:
TBH to spot cases like that, I'd turn on Checker's _concurrent semantics_ for the entire project (or at least for `frontendActions`): https://checkerframework.org/manual/#faq-concurrency

Reply:
Enabling this option may require a lot of changes unrelated to this PR.
Issue #276 created